### ISSUE TITLE
Creating WP Dev environment fails due to missing readable-string module

### ISSUE THREAD

**Issue by Viper007Bond** [10up/wp-local-docker-v2#19]
When creating a new environment that uses the development version of WordPress:

```
Resolving deltas: 100% (240581/240581), done.
Checking out files: 100% (2837/2837), done.
Checking USER ID
Host user is 0:0

> phantomjs-prebuilt@2.1.16 install /usr/src/app/node_modules/phantomjs-prebuilt
> node install.js

module.js:550
    throw err;
    ^

Error: Cannot find module 'readable-stream'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/concat-stream/index.js:1:78)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
```

`npm install` seems to work fine if I run it inside the `wordpress` directory using WSL.

**Comment by cmmarslender** [comment]
Are you still getting this error? Do you get it with the latest images (10updocker image update) ? I'm not able to reproduce this.

**Comment by Viper007Bond** [comment]
Yes, still getting this error. Latest everythging. I even switched from the recommended Node v8 to v10. I should make sure to note that I'm running this from Windows command prompt but I wouldn't think it'd matter as this is all supposed to be running inside of Docker, right?

```
> phantomjs-prebuilt@2.1.16 install /usr/src/app/node_modules/phantomjs-prebuilt
> node install.js

module.js:550
    throw err;
    ^

Error: Cannot find module 'readable-stream'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/concat-stream/index.js:1:78)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
npm WARN rollback Rolling back jsonfile@2.4.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/jsonfile'
npm WARN rollback Rolling back yauzl@2.4.1 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/extract-zip/node_modules/yauzl'
npm WARN rollback Rolling back throttleit@1.0.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/throttleit'
npm WARN rollback Rolling back hasha@2.2.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/hasha'
npm WARN rollback Rolling back kew@0.7.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/kew'
npm WARN rollback Rolling back fd-slicer@1.0.1 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/extract-zip/node_modules/fd-slicer'
npm WARN rollback Rolling back klaw@1.3.1 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/klaw'
npm WARN rollback Rolling back nan@2.9.2 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/nan'
npm WARN rollback Rolling back request-progress@2.0.1 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/request-progress'
npm WARN rollback Rolling back which@1.3.1 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/phantomjs-prebuilt/node_modules/which'
npm WARN rollback Rolling back progress@1.1.8 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/phantomjs-prebuilt/node_modules/progress'
npm WARN rollback Rolling back grunt-contrib-qunit@1.3.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/grunt-contrib-qunit'
npm WARN rollback Rolling back fs-extra@1.0.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/fs-extra'
npm WARN rollback Rolling back es6-promise@4.2.5 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/es6-promise'
npm WARN rollback Rolling back phantomjs-prebuilt@2.1.16 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/phantomjs-prebuilt/node_modules/progress'
npm WARN rollback Rolling back extract-zip@1.6.7 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/extract-zip/node_modules/yauzl'
npm WARN rollback Rolling back grunt-lib-phantomjs@1.1.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/usr/src/app/node_modules/grunt-lib-phantomjs/node_modules/rimraf'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! phantomjs-prebuilt@2.1.16 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the phantomjs-prebuilt@2.1.16 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/www/.npm/_logs/2019-02-05T01_50_49_169Z-debug.log
(node:19932) UnhandledPromiseRejectionWarning: Error: Command failed: docker run -t --rm -v C:\work\wpdev\wordpress-dev-localhost/wordpress:/usr/src/app -v wplocaldockerCache:/var/www/.npm 10up/wpcorebuild:latest npm install
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:653:13)
    at Object.downloadDevelop (C:\Users\viper\AppData\Roaming\nvm\v10.15.1\node_modules\wp-local-docker\src\wordpress.js:17:5)
(node:19932) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:19932) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by Viper007Bond** [comment]
I also have Docker set up in WSL as a client to talk to the Docker server in Windows (where it has to run) but it fails when trying to create the phpfm image:

```
 work/wpdev/regenerate-thumbnails-localhost  10updocker start
Locating project files for regenerate-thumbnails-localhost
Ensuring global network exists
 - Network exists
Ensuring global cache volume exists
 - Creating Volume
Ensuring global services are running
global_mysql_1 is up-to-date
global_gateway_1 is up-to-date
global_mailcatcher_1 is up-to-date
global_dns_1 is up-to-date
global_phpmyadmin_1 is up-to-date
Waiting for mysql...

Starting docker containers for regenerate-thumbnails-localhost
Creating network "regenerate-thumbnails-localhost_default" with the default driver
Creating regenerate-thumbnails-localhost_memcached_1 ... done
Creating regenerate-thumbnails-localhost_phpfpm_1        ... error
Creating regenerate-thumbnails-localhost_memcacheadmin_1 ... done

ERROR: for regenerate-thumbnails-localhost_phpfpm_1  Cannot start service phpfpm: OCI runtime create failed: container_linux.go:344: starting container process caused "process_linux.go:424: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/c/work/wpdev/regenerate-thumbnails-localhost/config/php-fpm/docker-php-ext-xdebug.ini\\\" to rootfs \\\"/var/lib/docker/overlay2/a6c4866fcbf5278c8955c4d08afef999151c3c265a106a133566d4ce76fe1e0f/merged\\\" at \\\"/var/lib/docker/overlay2/a6c4866fcbf5278c8955c4d08afef999151c3c265a106a133566d4ce76fe1e0f/merged/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ERROR: for phpfpm  Cannot start service phpfpm: OCI runtime create failed: container_linux.go:344: starting container process caused "process_linux.go:424: container init caused \"rootfs_linux.go:58: mounting \\\"/mnt/c/work/wpdev/regenerate-thumbnails-localhost/config/php-fpm/docker-php-ext-xdebug.ini\\\" to rootfs \\\"/var/lib/docker/overlay2/a6c4866fcbf5278c8955c4d08afef999151c3c265a106a133566d4ce76fe1e0f/merged\\\" at \\\"/var/lib/docker/overlay2/a6c4866fcbf5278c8955c4d08afef999151c3c265a106a133566d4ce76fe1e0f/merged/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
ERROR: Encountered errors while bringing up the project.

Environment configured for the following domains:
regenerate-thumbnails.localhost
regen.localhost
```

**Comment by tmdesigned** [comment]
I actually have the same issue. The 'readable_string' message is generated during the `npm install` of the dev WordPress. 

The error message is actually not that uncommon on the web. Most of the suggestions revolve around reinstalling node/npm after clearing out the AppData/Roaming/npm & npm-cache directories, due to corruption and/or permissions issues. This did not help, nor did changing versions, but I don't believe Docker is using that anyways. Uninstalling and reinstalling Docker did not resolve the issue either. 

Node 11.14, npm 6.7.0, Docker 18.09.2, Windows 10 Pro, fresh install of wp-local-docker.

**Comment by tmdesigned** [comment]
Forgot to mention, I'd be happy to help debug or try out potential solutions. Coming from VVV however and this is my first experience using Docker.

**Comment by tmdesigned** [comment]
@cmmarslender can you share a known working Windows configuration? i.e. node and npm version, which type of Docker installation, which shell, whether using admin, and anything else you can think of.

**Comment by cmmarslender** [comment]
@tmdesigned Sorry for delay. Needed to get windows environment back up to test. I was able to get this to work on windows 10 in powershell using node 8.11.4. I updated node to 10.16.0 and tried again and was successful with that version as well. 

You had a previous comment that this is typically an issue with node/npm cache. If you deleted the cache directories on the host machine, that wouldn't have helped the issue since this is run in a docker container and keeps the cache files on a docker volume. Try running `10updocker cache clear` [Docs Here](https://10up.github.io/wp-local-docker-docs/global-commands/) to delete and recreate the cache volume and hopefully that solves the issue.
